tests/unit_tests/core/router/identity.cc

```diff
@@ -85,4 +86,47 @@ BOOST_AUTO_TEST_CASE(ParseIdentityFailure)
         0);
 }
 
+BOOST_AUTO_TEST_CASE(ValidCreateRoutingKey)
+{
+  core::IdentityEx ident;
+  BOOST_CHECK(
+      ident.FromBuffer(m_AliceIdentity.data(), m_AliceIdentity.size()));
+  BOOST_CHECK_NO_THROW(core::CreateRoutingKey(ident.GetIdentHash()));
+  // Ensure a valid identity hash is returned
+  BOOST_CHECK(core::CreateRoutingKey(ident.GetIdentHash()).ToBase32().size());
+}
+
+BOOST_AUTO_TEST_CASE(CreateInvalidRoutingKey)
+{
+  kovri::core::IdentHash zeroed_ident;
+  BOOST_CHECK_THROW(core::CreateRoutingKey(nullptr), std::invalid_argument);
+  BOOST_CHECK_THROW(core::CreateRoutingKey(zeroed_ident), std::invalid_argument);
+}
+
+BOOST_AUTO_TEST_CASE(ValidFormatRoutingKey)
+{
+  core::IdentityEx ident;
+  BOOST_CHECK(
+      ident.FromBuffer(m_AliceIdentity.data(), m_AliceIdentity.size()));
+
+  BOOST_CHECK_NO_THROW(core::FormatRoutingKey(ident.GetIdentHash()));
+  const std::string formatted_key = core::FormatRoutingKey(ident.GetIdentHash());
```

Comment:
This case does not test for a valid format. You will have to supply an extra argument to the formatting function (a ptime object), and then process accordingly (ensuring that time given is properly formatted).

Reply:
I used `boost::gregorian::from_undelimited_string()` below, which I thought would throw when supplied an invalid ISO date string. If you prefer an explicit `ptime` argument, I can try to implement that in the next set of refactors.